### ISSUE TITLE
Permissions conflict prevents updating / installing gems 

### ISSUE THREAD

**Issue by itspomf** [oneclick/rubyinstaller2#106]
## What problems are you experiencing?

When attempting to update installed gems, an unspecified conflict in the permissions occurs, which produces the following output:

```
Updating installed gems
Updating ffi
ERROR:  While executing gem ... (Gem::Package::PathError)
    installing into parent path lib/2.0/ffi_c.so of D:/Ruby/lib/ruby/gems/2.4.0/gems/ffi-1.9.23-x64-mingw32 is not allowed
```

Even after removing the indicated directory, it will be recreated the next time an install or update is attempted and the conflict persists.

This may be related to the following Rubygems issues regarding the handling of symlinks:

* [Issue #1448](https://github.com/rubygems/rubygems/issues/1448) (closed)
* [Issue #1802](https://github.com/rubygems/rubygems/issues/1802) (closed)

### Note

This appears to affect the Ruby 2.5.0 and RubyInstaller-Head (2.6.0-dev) packages for RubyInstaller2 as well.

## Steps to reproduce

Simply use `gem update`.

## What's the output from `ridk version`?

```yaml
---
ruby:
  path: D:/Ruby
  version: 2.4.3
  platform: x64-mingw32
ruby_installer:
  package_version: 2.4.3-1
  git_commit: 36aac73
msys2:
  path: D:\Ruby\msys64
  title: MSYS2 64bit
  version: '20161025'
cc: x86_64-w64-mingw32-gcc (Rev1, Built by MSYS2 project) 7.3.0
sh: GNU bash, version 4.4.19(2)-release (x86_64-pc-msys)
os: Microsoft Windows [Version 6.1.7601]
```

MSYS2 and all packages have been updated as of the time of writing.

## Other environment details

Rubygems version 2.7.6

Ruby version `ruby 2.4.3p205 (2017-12-14 revision 61247) [x64-mingw32]`

**Comment by larskanis** [comment]
This bug was introduced in rubygems-1.7.6 and will be fixed in 1.7.7. See this [pull request](https://github.com/rubygems/rubygems/commit/35984510e0a3c192e66571b5bd1ce80f281ad0e8). I'll publish rubyinstaller-2.4.4 and 2.5.1 tomorrow with the above patch applied.

As an interim solution you can change the upper/lowercase notation of your ruby path. This should bypass this overly sensitive security check.

**Comment by itspomf** [comment]
Awesome news!  I'll give it a check in the next day or two and update the issue.

**Comment by itspomf** [comment]
This issue has been resolved with the Ruby 2.5.1-1 release.

**Comment by larskanis** [comment]
Thank you @itspomf for trying this out.
